File: LeetCode/1208. Get Equal Substrings Within Budget/solution.cpp
/**
 * 37 / 37 test cases passed.
 * Status: Accepted
 * Runtime: 8 ms
 */
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        int ans = 0;
        for (int l = 0, r = 0, sum = 0; r < s.size(); ++ r) {
            sum += abs(s[r] - t[r]);
            while (sum > maxCost) {
                sum -= abs(s[l] - t[l]);
                l ++;
            } 
            ans = max(ans, r - l + 1);
        }
        return ans;
    }
};

File: LeetCode/1208. Get Equal Substrings Within Budget/solution.py
"""
37 / 37 test cases passed.
Status: Accepted
Runtime: 88 ms
"""
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        cost = [abs(ord(s[i]) - ord(t[i])) for i in range(len(s))]
        start, ans, sum_range = 0, 0, 0
        for end in range(len(cost)):
            sum_range += cost[end]
            while sum_range > maxCost:
                sum_range -= cost[start]
                start += 1
            ans = max(ans, end - start + 1)
        return ans
